pkg/strategy/dca2/takeProfit.go

```diff
@@ -0,0 +1,42 @@
+package dca2
+
+import (
+	"context"
+
+	"github.com/c9s/bbgo/pkg/fixedpoint"
+	"github.com/c9s/bbgo/pkg/types"
+)
+
+func (s *Strategy) openTakeProfitOrders(ctx context.Context) error {
+	s.logger.Info("[DCA] open take profit orders")
+	takeProfitOrder := s.generateTakeProfitOrder(s.Short, s.Position)
+	createdOrders, err := s.OrderExecutor.SubmitOrders(ctx, takeProfitOrder)
+	if err != nil {
+		return err
+	}
+
+	for _, createdOrder := range createdOrders {
+		s.logger.Info("SUBMIT TAKE PROFIT ORDER ", createdOrder.String())
+	}
+
+	return nil
+}
+
+func (s *Strategy) generateTakeProfitOrder(short bool, position *types.Position) types.SubmitOrder {
+	takeProfitRatio := s.TakeProfitRatio
```

Comment:
you can move this to the func arg, so that you just pass s.takeProfitRatio from the caller